pkg/controller/common/util.go

```diff
@@ -279,6 +279,24 @@ var (
 	apiServerKey     *rsa.PrivateKey
 )
 
+// SizeMismatchError is an error returned if source size is larger than target
+type SizeMismatchError struct {
+	targetQuantity resource.Quantity
+	sourceQuantity resource.Quantity
+}
+
+func (e *SizeMismatchError) Error() string {
+	return fmt.Sprintf("target resources requests storage size %d is smaller than the source %d", e.targetQuantity.Value(), e.sourceQuantity.Value())
+}
+
+// NewSizeMismatchError returns a new SizeMismatchError
+func NewSizeMismatchError(target, source resource.Quantity) *SizeMismatchError {
```

Comment:
Any reasons to add the new error type? I'm not opposed to it, but at first glance seemed overcomplicated for a simple string.

Reply:
Yeah - that string was used in different pkgs, and needed to parameterize it with values,
So I thought it is a little nicer than a func with the sizes as params